examples/column/empty_interface.go

```diff
@@ -5,10 +5,10 @@ func main() {
 
 	var Num interface{} = -1
 
-	// 型が異なるので実装できない
+	// Numと1の型が異なるのでinvalidな式の評価になる

```

Comment:
```suggestion
	// Numと1の型が異なるのでinvalidな式になる

```
[Optional/取り入れても取り入れなくてもOKです]
invalidなのは式なので「評価」と言わなくてもいいかなと思いました。

Reply:
これってExpressionの定義は満たしていませんか？
ref: https://golang.org/ref/spec#Expression

playgroundで実行したところ、「invalid operation」というエラーメッセージだったので、評価の時点で失敗しているのかと考えており、「評価」という言葉を入れました。
ref: https://play.golang.org/p/C0UiMrGxV8E

Reply:
> これってExpressionの定義は満たしていませんか？
> ref: https://golang.org/ref/spec#Expression

はい、Expressionの定義は満たしていると思います。しかし、EBNF上Expressionの定義を満たすものが全て合法なExpressionと言うわけではなく、`Num + 1`はinvalidな式である、というように理解しています。(EBNFはあくまで仕様の一部）

> playgroundで実行したところ、「invalid operation」というエラーメッセージだったので、評価の時点で失敗していると考えており、「評価」という言葉を入れました。
> ref: https://play.golang.org/p/C0UiMrGxV8E

説明ありがとうございます！考え方理解できました。

自分の理解だとoperation(演算)とevaluation(評価)は別の概念です。operationはここでは`Num`というOperandと`1`というOperandから`Num + 1`という式を作ることを言うと思います。ピッタリした根拠はないかもしれませんが、

> Operators combine operands into expressions.

あたりとは整合的な解釈かなと思います。一方評価は例えば`1+(2*3)`のような式を`7`と「評価」することを言うと考えています。この理解に基づくと、`Num + 1`は評価する前からinvalidなので、「評価」の語はなくて良いかな、という考えでした。

自分のコメントの趣旨は↑のごとくなのですが、とはいえ仕様書で根拠づけた考えというわけでもなく微妙なところなので、Optionalにしておいた次第です。

Reply:
> 自分の理解だとoperation(演算)とevaluation(評価)は別の概念です。

この文でNobishiiさんの考えていることが理解できました。ありがとうございます。

「式」の定義に関して曖昧なところがあるため、コメントを厳密にすれば良いのではないかなと思いました。
具体的には、「演算」がエラーメッセージとして提示されているので、
`Numと1の型が異なるので、invalidな式の演算が原因でcompile errorになる`
という厳密な表現に変えます。併せて、Coversionの方も
`intとNumのUnderlying typeが異なるので、Coversionできないことが原因でcompile errorになる`
という表現に変えます。